lib/pbench/test/unit/server/test_cache_manager.py

```diff
@@ -485,260 +499,127 @@ def mock_run(args, **kwargs):
-                        'subdir1':
-                        {
-                            'details': <pbench.server.cache_manager.FileInfo object at 0x7fc4f1d20110>,
-                            'children':
-                            {
-                                'f1.txt': {'details': <pbench.server.cache_manager.FileInfo object at 0x7fc4f1dc51d0>},
-                                'subdir13':
-                                {
-                                    'details': <pbench.server.cache_manager.FileInfo object at 0x7fc4f1dc4990>,
-                                    'children':
-                                    {
-                                        'subdir111':
-                                        {
-                                            'details': <pbench.server.cache_manager.FileInfo object at 0x7fc4f1dc4310>,
-                                            'children':
-                                            {
-                                                'f1_sym': { 'details': <pbench.server.cache_manager.FileInfo object at 0x7fc4f1db2250>}
-                                                'f1.txt': { 'details': <pbench.server.cache_manager.FileInfo object at 0x7fc4f1db01d0>}
-                                            }
-                                        }
-                                    }
-                                },
-                                'subdir12':
-                                {
-                                    'details': <pbench.server.cache_manager.FileInfo object at 0x7fc4f1dc48d0>
-                                },
-                                'subdir11':
-                                {
-                                    'details': <pbench.server.cache_manager.FileInfo object at 0x7fc4f1d98850>
-                                }
-                            }
-                        }
-                    }
-                }
-            }
-            """
-
-            m.setattr(Tarball, "__init__", TestCacheManager.MockTarball.__init__)
-            m.setattr(Controller, "__init__", TestCacheManager.MockController.__init__)
-            tb = Tarball(tar, Controller(Path("/mock/archive"), cache, None))
-            tar_dir = TestCacheManager.MockTarball.create_tarball("tar_name")
-            tb.cache_map(tar_dir)
-            assert tb.cachemap["tar_name"]["details"].type == "directory"
+            assert c_map["details"].type == "symlink"
+            assert c_map["details"].name == "f1112_sym"
 
             # test get_info with a file path
-            file_info = tb.get_info(Path("tar_name/subdir1/f1.txt"))
+            file_info = tb.get_info(Path("dir_name/subdir1/f11.txt"))
             expected_info = {
-                "location": Path("/tmp/tar_name/subdir1/f1.txt"),
-                "name": "f1.txt",
-                "size": "1024",
+                "location": Path(f"{tar_dir}/subdir1/f11.txt"),
```

Comment:
Right...there are two issues here.  One is Webb's nit-pickiness, around using both `"dir_name"` and `f"{tar_dir}"` which _should_ refer to the same thing (but probably don't quite).  The other is dealing with the problem of relative vs. absolute paths.

The first is not really a big deal (although, if we can find a way to reconcile them, it would be nice/good).

The fact that second keeps coming up is starting to make me nervous.  References to items in the cache map really should be made with relative paths.  There are several reasons for this, but the important ones have to do with safety.  The fact that `location` ends up with an absolute path is a convenience based on an artifact of the `Path.resolve()` call.  We should consider editing that that string before storing it to remove the path to the base of the cache, so that the result is a relative path within the cache, so that, e.g., it could be used to do a traversal.